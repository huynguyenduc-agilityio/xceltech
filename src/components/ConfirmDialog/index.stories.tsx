import type { Meta, StoryObj } from '@storybook/react';

// Hooks
import { useConfirm } from '@/hooks';

// Components
import { Button } from '../common';
import ConfirmDialog from '.';

const meta: Meta<typeof ConfirmDialog> = {
  title: 'Components/ConfirmDialog',
  component: ConfirmDialog,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  decorators: [(Story) => <Story />],
};

export default meta;

type Story = StoryObj<typeof ConfirmDialog>;

const Template = () => {
  const confirm = useConfirm();

  const handleConfirmDialog = () =>
    confirm({
      title: 'Delete Item',
      confirmMessage: 'Are you sure you want to delete this item?',
      onConfirm: () => alert('Confirmed!'),
      onCancel: () => alert('Cancelled!'),
    });

  return (
    <>
      <Button onClick={handleConfirmDialog} className="px-4">
        Delete Item
      </Button>
      <ConfirmDialog />
    </>
  );
};

export const Default: Story = {
  render: Template,
};
