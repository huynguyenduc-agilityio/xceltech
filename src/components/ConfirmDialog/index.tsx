import { useConfirmDialogStore } from '@/stores';

import { Button, DialogContainer } from '../common';

const ConfirmDialog = () => {
  const { isOpen, title, confirmMessage, onConfirm, onCancel, close } =
    useConfirmDialogStore();

  if (!isOpen) return null;

  const handleConfirm = () => {
    onConfirm?.();
    close();
  };

  const handleCancel = () => {
    onCancel?.();
    close();
  };

  return (
    <DialogContainer
      isOpen={isOpen}
      title={title}
      description={confirmMessage}
      footer={
        <>
          <Button
            variant="outline"
            className="text-sm h-fit px-4 py-2"
            onClick={handleCancel}
          >
            Cancel
          </Button>
          <Button className="text-sm h-fit px-4 py-2" onClick={handleConfirm}>
            Confirm
          </Button>
        </>
      }
    />
  );
};

export default ConfirmDialog;
