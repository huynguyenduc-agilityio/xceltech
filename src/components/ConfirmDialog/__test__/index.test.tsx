import { render } from '@testing-library/react';

import ConfirmDialog from '..';

describe('ConfirmDialog components', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<ConfirmDialog />);

    expect(asFragment()).toMatchSnapshot();
  });
});
