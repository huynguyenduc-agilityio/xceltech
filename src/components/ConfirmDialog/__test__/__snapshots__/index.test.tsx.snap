// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog components should match snapshot 1`] = `<DocumentFragment />`;
